File: offer/68/68-1.js
var lowestCommonAncestor = function(root, p, q) {
    let res = root
    while(true){
        if(p.val < res.val && q.val < res.val){
            res = res.left
        }else if (p.val > res.val && q.val > res.val){
            res = res.right
        }else {
            break
        }
    }
    return res
};

File: offer/68/68-1.go
package main

import (
	. "../../TreeNode"
)

func lowestCommonAncestor1(root *TreeNode, p *TreeNode, q *TreeNode) *TreeNode {
	res := root
	for{
		if p.Val < root.Val && q.Val < root.Val{
			res = res.Left
		}else if p.Val > root.Val && q.Val > root.Val{
			res = res.Right
		}else {
			break
		}
	}
	return res
}
func main() {

}
